### DISCUSSION TITLE
<AccessIconOnly> Component

### DISCUSSION THREAD

**Discussion by shivarm** [Hyperfoil/Horreum#528] [Ideas]
I need to create a <AccessIconOnly> component but I'm not sure what is this going contain and How can I start with this?

**Comment by jesperpedersen** [comment]
There is no text element for `<AccessIconOnly>` - it is only the icon. Otherwise it is the same as `<AccessIcon>`

**Reply by shivarm** [reply]
you mean I need to change `<AccessIcon>` component into `<AccessIconOnly>`

**Reply by jesperpedersen** [reply]
No, create a new one

**Reply by shivarm** [reply]
what is this going to contain? and within same dir I have to create this one

**Reply by shivarm** [reply]
See this- https://github.com/Hyperfoil/Horreum/blob/a3a5d9f9a0b30713985261c4293a19a430109c44/webapp/src/components/AccessIcon.tsx#L5

**Reply by shivarm** [reply]
#### There is no text element for `<AccessIconOnly>`

```
import { Tooltip } from "@patternfly/react-core"
import { LockedIcon } from "@patternfly/react-icons"
import { Access } from "../api"

export default function AccessIconOnly({ access }: { access: Access }) {
    let color
    let explanation
    switch (access) {
        case 0: {
            color = "--pf-global--success-color--200"
            explanation = "Anyone can view this."
            break
        }
        case 1: {
            color = "--pf-global--warning-color--100"
            explanation = "Only authenticated (logged in) users can view this."
            break
        }
        case 2: {
            color = "--pf-global--danger-color--100"
            explanation = "Only users from the owning team can view this."
            break
        }
        default: {
            color = "--pf-global--icon--Color--light"
            explanation = "Unknown"
            break
        }
    }
    return (
        <>
            <LockedIcon style={{ fill: "var(" + color + ")" }} />
            {"\u00A0"}
            <Tooltip content={explanation}>
                <span>{}</span>
            </Tooltip>
        </>
    )
}
```

**Reply by jesperpedersen** [reply]
`<AccessIconOnly>` should not have a text element

**Reply by johnaohara** [reply]
Do we want to introduce a new component (which is mostly duplicated) or just not render the `text` component if text is not set?

**Reply by jesperpedersen** [reply]
Either way - what do you prefer ?

**Reply by johnaohara** [reply]
less duplication

**Reply by jesperpedersen** [reply]
@shivam-sharma7 Can you look into this ?

**Reply by shivarm** [reply]
@jesperpedersen I don't understand? What we need to implement in this component

**Reply by jesperpedersen** [reply]
Use `<AccessIcon>` but don't render the text if it isn't set

**Reply by shivarm** [reply]
Then my previous changes were okay in this case.

**Reply by jesperpedersen** [reply]
No, we don't want the "Public" / "Private" text in these cases

**Reply by shivarm** [reply]
Okay
